ocaml/utils/import_info.mli

```diff
@@ -43,8 +45,39 @@ val cu : t -> CU.t
 
 val crc : t -> string option
 
-val crc_with_unit : t -> (CU.t * string) option
-
 val has_name : t -> name:CU.Name.t -> bool
 
 val dummy : t
+
+(** The preferred API to use for interface imports. An interface import might be
+    a parameter, in which case it has a CRC but no [CU.t] (since a [CU.t] is for
+    an implementation). *)
+module Intf : sig
+  type nonrec t = t
+
+  val create : CU.Name.t -> CU.t option -> crc:Digest.t option -> t
```

Comment:
It would definitely be better to have things be better tied to their purpose, yeah. Should those be the three `create` functions? `create_normal`, `create_parameter`, and `create_alias`? (I didn't like `create_normal` before but that was when there wasn't anything to suggest what the "abnormal" cases were.)

Reply:
Hmm ... I'm not sure what to write for the other call sites to `Import_info.create*` in this case. Is it _only_ for aliases that we ever don't have a CRC?

Reply:
I think that `Intf` should have:
```ocaml
val create_ordinary : CU.Name.t -> CU.t -> Digest.t -> t
val create_parameter : CU.Name.t -> Digest.t -> t
val create_alias : CU.Name.t -> t
```
and that it should really be a variant with three cases under the hood. I think it would be worth, in this PR, already having an `Import_info.t` whose implementation looked like:
```ocaml
type t =
  | Intf of intf
  | Impl of impl
```
so that you can already start implementing the functionality for `intf` and `impl` separately.

For the other calls to `Import_info.create*`: in some cases they are using a `Consistbl` for the typing imports of a module, so that a missing digest means an alias, in other cases they are using a `Consistbl` for the flambda imports of a module, so that a missing digest just means we never read the `.cmx`. You should make them use your `Intf` or `Impl` modules as appropriate and both of those modules should have the appropriate create functions for the case it is intended for.

Reply:
Okay, added these functions for `Intf` and similarly added `create_loaded` and `create_unloaded` to `Impl`, and changed the type definitions around so they just store the arguments to the `create` functions (and then, as suggested, `t` is just a disjunction of `intf` and `impl`).